great_expectations/experimental/datasources/filesystem_data_asset.py

```diff
@@ -55,13 +63,28 @@ class Config:
 
         extra = pydantic.Extra.allow
 
+    def __init__(self, **data):
+        super().__init__(**data)
+        self._regex_parser = RegExParser(
+            regex_pattern=self.regex,
+            unnamed_regex_group_prefix=self._unnamed_regex_param_prefix,
+        )
+
+        self._all_group_name_to_group_index_mapping = (
+            self._regex_parser.get_all_group_name_to_group_index_mapping()
+        )
+        self._all_group_index_to_group_name_mapping = (
+            self._regex_parser.get_all_group_index_to_group_name_mapping()
+        )
+        self._all_group_names = self._regex_parser.get_all_group_names()
```

Comment:
I think you could do all these with `validator` methods, and then you wouldn't need this custom `__init__`.
🤔 
I think this is why the CI mypy said those ignores were no longer needed.

Will update this comment with a example implementation.

Reply:
Actually, I don't think we can do these with `pydantic.PrivateAttr()`.
I think we should make these normal attributes and then we can use validators and get ride of the custom init but I don't want to block the PR on this.

I'll followup with you and maybe we can pair on this.